api-server/scanner-daemon/src/config.rs

```diff
@@ -46,6 +46,30 @@ pub struct ApiServerScannerArgs {
     /// RPC password (either provide a username and password, or use a cookie file. You cannot use both)
     #[clap(long)]
     pub rpc_password: Option<String>,
+
+    /// Postgres host
+    #[clap(long, default_value = "localhost")]
+    pub postgres_host: String,
+
+    /// Postgres port
+    #[clap(long, default_value = "5432")]
+    pub postgres_port: u16,
+
+    /// Postgres user
+    #[clap(long, default_value = "postgres")]
+    pub postgres_user: String,
+
+    /// Postgres password
+    #[clap(long)]
+    pub postgres_password: Option<String>,
+
+    /// Postgres database
+    #[clap(long)]
+    pub postgres_database: Option<String>,
+
+    /// Postgres max connections
+    #[clap(long, default_value = "10")]
+    pub postgres_max_connections: u32,
```

Comment:
Please make an attempt to group postgres-related configuration to in its own struct. I think clap supports composable configuration, not sure though. 

Please note that we DO NOT want use commands for this. I know commands can do it, but it's the wrong tool. It would be sufficient to just be able to put these in a struct then throw the struct in here.

IF that works, then it would be great if we could have tree-like validation, where postgres settings only work if the database type is postgres. But that comes later, no need to spend too much time on it. Let's start with the struct. 

If that struct works, then please put it in api-server-common, and use it for both the scanner-daemon and the web-server.

Reply:
The new changes have a separate struct for Postgres, and are parsed via the `default()`. If we want to be optional vs other backing storage types as discussed in the meeting, I'm not sure if Clap supports the Enum like you mentioned so you may have to detect manually after `parse()` the top-level?

Reply:
If you want to use an enum you can actually use a subcommand like: https://github.com/mintlayer/mintlayer-core/blob/e3c81965ceed2aa774793809c7d2e7ecccad5923/node-lib/src/options.rs#L44-L52